### PR TITLE
`Plugin` Minor UI Enhancements.

### PR DESCRIPTION
**Pull request by kemzops**
### Two Enhancements :
- Minor UI Enhancement.
image

- `config_spec` to support the settings gui. 
image

### CODE REVIEW

File: plugins/plugin_manager/plugin_view.lua [hunk 1]
```diff
@@ -166,17 +161,24 @@ function PluginView:draw()
     return self:draw_loading_screen(self.progress and self.progress.label, self.progress and self.progress.percent)
   end
 
+  renderer.draw_rect(self.position.x, self.position.y, self.size.x, 1, style.dim)
```
Comment (adamharrison) [inline plugins/plugin_manager/plugin_view.lua:164]: I am honestly not crazy about the lines.

File: plugins/plugin_manager/plugin_view.lua [hunk 2]
```diff
@@ -166,17 +161,24 @@ function PluginView:draw()
...
 
   local ox, oy = self:get_content_offset()
   oy = oy + lh * self.offset_y
 
   local x, y = ox + style.padding.x, oy
   for i, v in ipairs(self.plugin_table_columns) do
     common.draw_text(style.font, style.accent, v, "left", x, self.position.y, self.widths[i], lh)
-    x = x + self.widths[i] + style.padding.x
+    renderer.draw_rect(x + self.widths[i], self.position.y, 1, self.size.y, style.dim)
```
Comment (adamharrison) [inline plugins/plugin_manager/plugin_view.lua:172]: And here.

File: plugins/plugin_manager/plugin_view.lua [hunk 3]
```diff
@@ -166,17 +161,24 @@ function PluginView:draw()
...
+    x = x + self.widths[i] + style.padding.x + 2
   end
 
+  renderer.draw_rect(self.position.x, self.position.y + lh * self.offset_y, self.size.x, 1, style.dim)
```
Comment (adamharrison) [inline plugins/plugin_manager/plugin_view.lua:176]: And here.

File: plugins/plugin_manager/plugin_view.lua [hunk 4]
```diff
@@ -166,17 +161,24 @@ function PluginView:draw()
...
+  
+  local x = self.position.x + self.size.x - style.padding.x - style.font:get_width("Search")
+  common.draw_text(style.font, style.dim, "Search: Ctrl+F", "right", x, self.position.y, style.font:get_width("Ctrl+F"), lh)
```
Comment (adamharrison) [inline plugins/plugin_manager/plugin_view.lua:179]: This, I like. But you should probably use `keymap.get_binding` to display this. Also, why `- style.font:get_width("Search")`? Shouldn't you just pile it all into the second line, and make the width passed to `draw_text` the width of the whole string?

**Comment by kemzops** [comment]
@adamharrison thanks for the review, i tried to make it a table, and it worked but it wasn't very good to the eyes, these lines make a good separation between the rows and columns to make it more visible and easier to scroll throw.

 `style.font:get_width("Search")` is just a quick hack? to make it visible with fixed place.
 `keymap.get_binding` im really learning by doing, i didn't read any docs for this so this just worked, if you want i can try to enhance this more and add more things to it since i pushed this PR initially to check if my way of doing things work or not.

**Comment by kemzops** [comment]
i guess no.

**Comment by adamharrison** [comment]
Hey; I'm a busy guy. Just because I don't respond in a day or two doesn't mean I'm not interested.

I'm not crazy about the lines, but I wouldn't mind some more enhancements with the search stuff stuff in the top right; and the keybinding thing. Do you think you could do that bit?

**Comment by kemzops** [comment]
I respect that, because you did not reply i thought you are not interested in this PR at all, im sorry for the rush, my bad.

**Comment by kemzops** [comment]
the last commit i removed the search text instead perform the find command by default when opening the plugin manager.

image
image

**Comment by kemzops** [comment]
if this idea is good just let me know so i can implement it correctly, this is just a "test" commit for it. :smiley:

**Pull request closed without merging**